src/input/input.tsx

```diff
@@ -96,29 +97,43 @@ const Component = (
      */
     const updatedValue = value ? convertInputString(value) : value;
 
+    const renderInputElement = () => {
+        return (
+            <>
+                <InputElement
+                    data-testid="input"
+                    ref={elementRef}
+                    disabled={disabled}
+                    value={updatedValue}
+                    onChange={handleChange}
+                    type={type}
+                    readOnly={readOnly}
+                    {...otherProps}
+                />
+                {shouldShowClear() && (
+                    <ClearContainer onClick={handleClear} type="button">
+                        <ClearIcon aria-hidden />
+                    </ClearContainer>
+                )}
+            </>
+        );
+    };
+
     return (
-        <InputWrapper
-            $disabled={disabled}
-            $error={error}
-            $readOnly={readOnly}
-            className={className}
-        >
-            <InputElement
-                data-testid="input"
-                ref={elementRef}
-                disabled={disabled}
-                value={updatedValue}
-                onChange={handleChange}
-                type={type}
-                readOnly={readOnly}
-                {...otherProps}
-            />
-            {shouldShowClear() && (
-                <ClearContainer onClick={handleClear} type="button">
-                    <ClearIcon aria-hidden />
-                </ClearContainer>
+        <>
+            {noWrapper ? (
+                renderInputElement()
```

Comment:
I think this should still wrap the input and clear button in a basic div that renders them on the same line